main.c

```diff
@@ -23,12 +23,15 @@
 #define SAMPLES 5
 #define TIMES 5
 #define BYTES_PER_GB (1024*1024*1024LL)
-#define SIZE (1*BYTES_PER_GB)
-#define PAGE_SIZE (1<<12)
+#define MAX_SIZE (1*BYTES_PER_GB)
+#ifndef PAGE_SIZE
+# define PAGE_SIZE (1<<12)
+#endif
+static unsigned long long SIZE = MAX_SIZE;
 
 // This must be at least 32 byte aligned to make some AVX instructions happy.
 // Have PAGE_SIZE buffering so we don't have to do math for prefetching.
-char array[SIZE + PAGE_SIZE] __attribute__((aligned (32)));
+char array[MAX_SIZE + PAGE_SIZE] __attribute__((aligned (32)));
```

Comment:
I don't see that we need the + PAGE_SIZE here unless there is some deeper purpose from the CPU vs. memory architecture.